presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1424,6 +1470,68 @@ else if (materializedViewStatus.isPartiallyMaterialized()) {
             return SqlFormatterUtil.getFormattedSql(unionQuery, sqlParser, Optional.empty());
         }
 
+        /**
+         * Returns materialized view status, considering filter conditions from query
+         */
+        private MaterializedViewStatus getMaterializedViewStatus(QualifiedObjectName materializedViewName, Optional<Scope> scope, Table table)
```

Comment:
Can this be moved to materializedViewUtils class?

Reply:
Not trivially AFAICT- the `getMaterializedViewStatus` in `MaterializedViewQueryOptimizer` and the one in `StatementAnalyzer` do quite a few things differently. `StatementAnalyzer` for example requires getting mappings from the view/rewritten query to base table, while `MaterializedViewQueryOptimizer` has access to the base query and doesn't need to worry about them.

I could extract a couple of short helper methods, but not sure how much that would help readability.